RecoJets/JetProducers/src/PileupJetIdAlgo.cc

```diff
@@ -195,6 +171,17 @@ void PileupJetIdAlgo::set(const PileupJetIdentifier & id)
 }
 
 // ------------------------------------------------------------------------------------------
+
+std::vector<float> PileupJetIdAlgo::getMVAvars(std::vector<std::string> varList)
+{
+        std::vector<float> vars;
+        for(std::vector<std::string>::iterator it=varList.begin(); it!=varList.end(); ++it) {
+            std::pair<float *,float> var = variables_.at((*it).c_str());
+            vars.push_back( *var.first );
+        }
+        return vars;
+}
```

Comment:
This method needs some performance improvements. One should always avoid having a function return a large object like a vector. You could change this function to:
```
float PileupJetIdAlgo::getMVAvars(const std::unique_ptr<const GBRForest> &reader, const std::vector<std::string> &varList)
{
  // Same code as you have except for return
  return (reader_->GetClassifier(vars.data()));
}
```